CRM/Core/BAO/MessageTemplate.php

```diff
@@ -83,6 +83,27 @@ public static function setIsActive($id, $is_active) {
    * @return object
    */
   public static function add(&$params) {
+    // System Workflow Templates have a specific wodkflow_id in them but normal user end message templates don't
+    // If we have an id check to see if we are update, and need to check if original is a system workflow or not.
+    if (!empty($params['id']) && !empty($params['check_permissions'])) {
```

Comment:
@eileenmcnaughton my understanding was that the idea was that it would have to be edit message templates + either system or user edit permissions. i.e. getting away from just having edit message templates.

Reply:
@ajesamson @guanhuan this was a Compucorp PR so comments very welcome

@seamuslee001 see https://github.com/civicrm/civicrm-core/pull/11974 - " This helps to further limit which of the templates a user can view and edit, while retaining exiting edit message templates permission for backward compability."

My understanding is that 'edit message templates' should work as it always has - to use more granular this should be removed & the more granular used instead

Reply:
@seamuslee001 What @eileenmcnaughton  said is correct. It is not meant to be 

> edit message templates + either system workflow or user-driven permissions

The essence of leaving `edit message templates` permission is for backward compatibility. Users are notified on [upgrade](https://github.com/civicrm/civicrm-core/blob/master/CRM/Upgrade/Incremental/php/FiveThree.php#L49) (as [suggested](https://github.com/civicrm/civicrm-core/pull/11974#issuecomment-382802799) by @colemanw due to some implementation complexity) of the new permissions. They are then expected; at their own discretion, to disable `edit message templates` while assigning either `system workflow` or `user-driven` message templates permission (or both) to users as applicable to their system.

cc @mickadoo

Reply:
yep understand now, i'm working on an update just now, that should implement that concept as well.